### PR TITLE
launcher: Change retry policy to hourly and add jitter to refresh time

### PR DESCRIPTION
**Pull request by alexmwu**
We want the retry to occur ~hourly (with jitter), as this will match the normal refreshed requests. This allows the launcher to provide a token even with long service outages.

Also, jitter on the refresh duration allows smearing of requests in the event many clients come online at similar times after an outage. Without this, load on the service side would be less balanced over time.

### CODE REVIEW

**Review by jessieqliu — APPROVED** [review]

File: launcher/container_runner.go [hunk 1]
```diff
@@ -427,15 +409,45 @@ func (r *ContainerRunner) fetchAndWriteTokenWithRetry(ctx context.Context,
...
-// defaultRetryPolicy retries with:
-// initial interval of 30s, multiplication factor of 1.5
-// randomization factor of 0.5, max interval of 3m, and
-// max elapsed time of 10m.
+func getNextRefreshFromExpiration(expiration time.Duration) time.Duration {
+	diff := defaultRefreshJitter * float64(expiration)
+	center := defaultRefreshMultiplier * float64(expiration)
+	minRange := center - diff
+	return time.Duration(minRange + rand.Float64()*2*diff)
+
```
Comment (jkl73) [inline launcher/container_runner.go:417]: extra space

File: launcher/container_runner.go [hunk 2]
```diff
@@ -427,15 +409,45 @@ func (r *ContainerRunner) fetchAndWriteTokenWithRetry(ctx context.Context,
 	return nil
 }
 
-// defaultRetryPolicy retries with:
-// initial interval of 30s, multiplication factor of 1.5
-// randomization factor of 0.5, max interval of 3m, and
-// max elapsed time of 10m.
+func getNextRefreshFromExpiration(expiration time.Duration) time.Duration {
```
Comment (jkl73) [inline launcher/container_runner.go:412]: Ideally, should have a unit test to test the returned time is always smaller than the expiration time.

**Review by jkl73 — APPROVED** [review]

File: launcher/container_runner.go (hunk 2 above)
Comment (alexmwu) [reply launcher/container_runner.go]: done. lmk if the test looks good to you.

**Review by michael-pregman — APPROVED** [review]

File: launcher/container_runner.go (hunk 2 above)
Comment (jkl73) [reply launcher/container_runner.go]: LGTM

**Pull request merged**